Mathlib/Tactic/Linter/Multigoal.lean

```diff
@@ -0,0 +1,158 @@
+/-
+Copyright (c) 2024 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import Lean.Elab.Command
+import Lean.Linter.Util
+
+/-!
+#  The "multiGoal" linter
+
+The "multiGoal" linter emits a warning where there is more than a single goal in scope.
```

Comment:
Re-re-thinking this wording, I am not completely happy: tactics creating side goals are fine. The point is when there are multiple active goals, but a tactic only acts on one of them. What do you think about something like the following?

```
The "multiGoal" linter emits a warning when there is more than a single goal in scope, but a tactic only acts on one of them.

As mandated by the mathlib style guide, when new goals arise as side conditions or steps,
they should be focused. It is common for tactics to split one goal into several steps, or to leave side conditions to be checked. In these cases, goals should be focused until there is only one active goal at a time. Typically, this focusing is achieved by the `cdot`: `·`, but, the `focus` or `on_goal` tactics (for example) serve a similar purpose.

The only exception to this rule are tactic combinators like `all_goals`, intended to run one or several tactics on several goals at once.
```

Reply:
I am in fact not fully sure if the first sentence is correct: would the following situation error?
-- one goal before
tactic1 -- two goals now
tactic2 -- closes both

In any case, can you add a test for something like this? My last adaptation PR contained one.

Reply:
You are right: there is an extra exception which is a tactic that starts with more than one open goals and the tactic closes *all* goals.  I.e., the linter shuts up if no goals are left.  I added this to the description.

Reply:
Thanks, this definitely helps! Do you have the energy to add a test for such a situation? (Would that be interesting? My motivation is to ensure these edge cases are not changed accidentally, and adding a test is an easy way to do so. Mathlib proofs change over time.)